codecs/avif/enc/avif_enc.cpp

```diff
@@ -29,34 +29,38 @@ val encode(std::string buffer, int width, int height, AvifOptions options) {
   int depth = 8;
   avifPixelFormat format;
   switch (options.subsample) {
-  case 0:
-    format = AVIF_PIXEL_FORMAT_YUV420;
-    break;
-  case 1:
-    format = AVIF_PIXEL_FORMAT_YUV422;
-    break;
-  case 2:
-    format = AVIF_PIXEL_FORMAT_YUV444;
-    break;
+    case 0:
+      format = AVIF_PIXEL_FORMAT_YUV420;
+      break;
+    case 1:
+      format = AVIF_PIXEL_FORMAT_YUV422;
+      break;
+    case 2:
+      format = AVIF_PIXEL_FORMAT_YUV444;
+      break;
   }
-  avifImage *image = avifImageCreate(width, height, depth, format);
 
-  uint8_t *rgba = (uint8_t *)buffer.c_str();
+  avifImage* image = avifImageCreate(width, height, depth, format);
 
-  avifImageAllocatePlanes(image, AVIF_PLANES_RGB);
-  avifImageAllocatePlanes(image, AVIF_PLANES_A);
+  uint8_t* rgba = (uint8_t*)buffer.c_str();
+
+  avifRGBImage srcRGB;
+  avifRGBImageSetDefaults(&srcRGB, image);
+  avifRGBImageAllocatePixels(&srcRGB);
 
   for (int y = 0; y < height; y++) {
     for (int x = 0; x < width; x++) {
       int pixelOffset = y * width + x;
-      image->rgbPlanes[0][pixelOffset] = rgba[pixelOffset * 4 + 0];
-      image->rgbPlanes[1][pixelOffset] = rgba[pixelOffset * 4 + 1];
-      image->rgbPlanes[2][pixelOffset] = rgba[pixelOffset * 4 + 2];
-      image->alphaPlane[pixelOffset] = rgba[pixelOffset * 4 + 3];
+      uint8_t* pixel = &srcRGB.pixels[(4 * x) + (srcRGB.rowBytes * y)];
```

Comment:
Not related to the build system, but it appears that this loop just copies bytes in original order without any reordering or recomputation? If so, can it be replaced entirely with `memcpy` or similar? Or, maybe, there is a method to create `avifRGBImage` from existing pointer to avoid copying data altogether?

Reply:
I wasn’t sure if they word-align rows or something, so technically I think `rowBytes` could be greater than `width * 4`, but not 100% sure. I was mostly following the example code in the AVIF repo.

Reply:
I just found my similar comment from the original PR, looks like you should be able to assign pixels directly: https://github.com/GoogleChromeLabs/squoosh/pull/722#discussion_r419785905

Here's another relevant comment from AVIF.h: https://github.com/AOMediaCodec/libavif/blob/01ee4caa5ab52b2f6f3e500fd3a0f70e0099b031/include/avif/avif.h#L365-L370